spec-old-api/02-integration/05-proxy/09-balancer_spec.lua

```diff
@@ -235,234 +237,396 @@ local function http_server(timeout, host, port, counts, test_log)
+    return upstream_name
+  end
+
+  patch_upstream = function(upstream_name, data)
+    assert.same(200, api_send("PATCH", "/upstreams/" .. upstream_name, data))
+  end
+
+  get_upstream_health = function(upstream_name)
+    local path = "/upstreams/" .. upstream_name .."/health"
+    local status, body = api_send("GET", path)
+    assert.same(200, status)
+    return body
+  end
+
+  do
+    local port = FIRST_PORT
+    gen_port = function()
+      port = port + 1
+      return port
+    end
+  end
+
+  add_target = function(upstream_name, host, port, data)
+    port = port or gen_port()
+    local req = utils.deep_copy(data) or {}
+    req.target = req.target or utils.format_host(host, port)
+    req.weight = req.weight or 10
+    local path = "/upstreams/" .. upstream_name .. "/targets"
+    assert.same(201, api_send("POST", path, req))
+    return port
+  end
+
+  add_api = function(upstream_name)
+    local api_name = gen_sym("api")
+    local api_host = gen_sym("host")
+    assert.same(201, api_send("POST", "/apis", {
+      name = api_name,
+      hosts = api_host,
+      upstream_url = "http://" .. upstream_name,
+    }))
+    return api_host, api_name
+  end
+
+  patch_api = function(api_name, new_upstream)
+    assert.same(200, api_send("PATCH", "/apis/" .. api_name, {
+      upstream_url = new_upstream
+    }))
+  end
+
+  add_plugin = function(api_name, body)
+    local path = "/apis/" .. api_name .. "/plugins"
+    local status, plugin = assert(api_send("POST", path, body))
+    assert.same(status, 201)
+    return plugin.id
+  end
+
+  delete_plugin = function(api_name, plugin_id)
+    local path = "/apis/" .. api_name .. "/plugins/" .. plugin_id
+    assert.same(204, api_send("DELETE", path, {}))
   end
-  api_client:close()
-  return res.status
+end
+
+
+local function truncate_relevant_tables()
+  helpers.dao.apis:truncate()
+  helpers.dao.upstreams:truncate()
+  helpers.dao.targets:truncate()
+  helpers.dao.plugins:truncate()
+end
+
+
+local function poll_wait_health(upstream_name, localhost, port, value)
+  local hard_timeout = 300
+  local expire = ngx.now() + hard_timeout
+  while ngx.now() < expire do
+    local health = get_upstream_health(upstream_name)
+    for _, d in ipairs(health.data) do
+      if d.target == localhost .. ":" .. port and d.health == value then
+        return
+      end
+    end
+    ngx.sleep(0.01) -- poll-wait
+  end
+end
+
+
+local function file_contains(filename, searched)
+  local fd = assert(io.open(filename, "r"))
+  for line in fd:lines() do
+    if line:find(searched, 1, true) then
+      fd:close()
+      return true
+    end
+  end
+  fd:close()
+  return false
 end
 
 
 local localhosts = {
   ipv4 = "127.0.0.1",
-  ipv6 = "0000:0000:0000:0000:0000:0000:0000:0001",
+  ipv6 = "[0000:0000:0000:0000:0000:0000:0000:0001]",
+  hostname = "localhost",
 }
 
 
-for ipv, localhost in pairs(localhosts) do
+dao_helpers.for_each_dao(function(kong_config)
```

Comment:
minor but just as a reminder: while we still iterate over the DBs in the tests, and because this helper comes from `spec` (and not `old-spec`), then we should use the new iterator provided by `spec/helpers.lua` and meant to replace this callback approach. The iterator is easier to use and also more flexible (it allows one to easily run only the `postgres` or `cassandra` version of the tests by specifying an argument.

Since this is the old-spec suite that is bound to disappear, it may be ok to use it there for now...